File: 94-binary-tree-inorder-traversal/Solution.py

from typing import List, Optional


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

        
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        
        result = []
        self.traveler(root, result)

        return result

    def traveler(self, root: Optional[TreeNode], result: List[int]):

        if root is not None:
            self.traveler(root.left, result)
            result.append(root.val)
            self.traveler(root.right, result)

root = TreeNode(1, TreeNode(2), TreeNode(3))

obj = Solution()

print(obj.inorderTraversal(root))

File: 94-binary-tree-inorder-traversal/Solution.java
import java.util.*;

class Solution {

    public static void main(String[] args) {
        
        TreeNode root = new TreeNode(1, new TreeNode(2), new TreeNode(3));

        System.out.println("--------------------");
        System.out.println(inorderTraversal(root));
    }


    public static List<Integer> inorderTraversal(TreeNode root) {
        
         List<Integer> result = new ArrayList<>();

        fillResult(root, result);

        return result;
    }
    
    public static void fillResult(TreeNode root, List<Integer> result){
        
        if(root != null) {
            fillResult(root.left, result);
            result.add(root.val);
            fillResult(root.right, result);
        }
    }
}